src/attributes/diagnostics.md

```diff
@@ -424,36 +467,59 @@ When used on a function in a trait implementation, the attribute does nothing.
 
 ## The `diagnostic` tool attribute namespace
 
+r[attributes.diagnostics.namespace]
```

Comment:
Hm, I'm on the fence here, since the namespace is singular `diagnostic`. I realize this is in the "diagnostics" section, and being inconsistent with the rest of the section could be more trouble. I'm just a little concerned about seeing `diagnostics.namespace` could confuse a reader, since it is a subtle distinction. WDYT?

Reply:
I'm thinking the section should be changed to `diagnostic` because I've been otherwise favouring the singular form except for lists and when the construct is plural in the language.